### DISCUSSION TITLE
Is there anyway to update IL instructions to use new local variable indexes?

### DISCUSSION THREAD

**Discussion by Uchiha1391** [jbevain/cecil#768] [Q&A] [answered]
I am trying to inject a method into another method. Problem is the local variables indexes. Example-

```
 public void Method1()
    {
        var tes = "anything";

    }

    public void Method2()
    {
        var spring = 40;

    }
```
I want to inject method1 into method2 but the problem is it throws invalid IL code because for method1 local variable at index 0 is of type string and for method2 local variable at index 0 is of type int. What I do is I fetch method1 methodDefinition and takes its instructuions through-

```
  var Method1_Instructions= MethodDefinition.Body.Instructions.ToList();// method1 methodDefinition
var Method1_localVariables= Method1_MethodDefinition.Body.Variables
   for (var I = 0; I < Method1_localVariables.Count; I++)
                    {
                        Method2_MethodDefinition.Body.Variables.Insert(I,
                            Method1_localVariables[I]); 

                    }
```
I add the local variables of method1 to method2 variables list and also insert instructions before it. But I know its wrong IL code because of the local variables index issue. So, How can I fix it without writing the IL code myself?

**Comment by jbevain** [comment, marked as answer]
Hello,

I'm afraid there's nothing built-in. You're going to have to fix the IL yourself to merge the bodies.
